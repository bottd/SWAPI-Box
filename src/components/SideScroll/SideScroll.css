.SideScroll {
  color: #FFF;
  overflow: hidden;
  text-align: justify;
}

.scrollText {
  color: #fdde09;
  font-size: 32px;
  line-height: 1.5;
  margin: 5px;
  position: aboslute;
  text-align: center;
}

.scroll {
  animation: scroll 60s infinite;
  position: relative;
}

@keyframes scroll {
  0% {
    top: 85vh;
  }
  100% {
    top: -110vh;
    display: none;
  }
}

@media screen and (max-width: 1000px) {
  .SideScroll {
    height: 86vh;
  }
}
