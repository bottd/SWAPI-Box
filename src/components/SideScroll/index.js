import React from "react";
import PropTypes from "prop-types";
import "./SideScroll.css";

const SideScroll = props => {
  const { title, opening_crawl, release_date } = props.episodeData;
  return (
    <div className="SideScroll">
      <div className="scroll">
        <p className="scrollText">{opening_crawl}</p>
        <p className="scrollText">{title}</p>
        <p className="scrollText">{release_date}</p>
      </div>
    </div>
  );
};

SideScroll.propTypes = {
  episodeData: PropTypes.objectOf(PropTypes.string)
};

export default SideScroll;
