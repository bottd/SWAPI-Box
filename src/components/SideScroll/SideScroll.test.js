import React from "react";
import SideScroll from "./index";
import { shallow } from "enzyme";

describe("SideScroll", () => {
  let wrapper;
  const exampleData = {
    title: 'A New Hope',
    opening_crawl: 'Opening crawl text here',
    release_date: '1977-05-25'
  };

  beforeEach(() => {
    wrapper = shallow(<SideScroll episodeData={exampleData}/>);
  });

  it("Should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

});
