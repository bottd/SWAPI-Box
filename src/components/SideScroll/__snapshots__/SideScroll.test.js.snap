// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideScroll Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideScroll
    episodeData={
      Object {
        "opening_crawl": "Opening crawl text here",
        "release_date": "1977-05-25",
        "title": "A New Hope",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="scroll"
      >
        <p
          className="scrollText"
        >
          Opening crawl text here
        </p>
        <p
          className="scrollText"
        >
          A New Hope
        </p>
        <p
          className="scrollText"
        >
          1977-05-25
        </p>
      </div>,
      "className": "SideScroll",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="scrollText"
          >
            Opening crawl text here
          </p>,
          <p
            className="scrollText"
          >
            A New Hope
          </p>,
          <p
            className="scrollText"
          >
            1977-05-25
          </p>,
        ],
        "className": "scroll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Opening crawl text here",
            "className": "scrollText",
          },
          "ref": null,
          "rendered": "Opening crawl text here",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A New Hope",
            "className": "scrollText",
          },
          "ref": null,
          "rendered": "A New Hope",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1977-05-25",
            "className": "scrollText",
          },
          "ref": null,
          "rendered": "1977-05-25",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="scroll"
        >
          <p
            className="scrollText"
          >
            Opening crawl text here
          </p>
          <p
            className="scrollText"
          >
            A New Hope
          </p>
          <p
            className="scrollText"
          >
            1977-05-25
          </p>
        </div>,
        "className": "SideScroll",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="scrollText"
            >
              Opening crawl text here
            </p>,
            <p
              className="scrollText"
            >
              A New Hope
            </p>,
            <p
              className="scrollText"
            >
              1977-05-25
            </p>,
          ],
          "className": "scroll",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Opening crawl text here",
              "className": "scrollText",
            },
            "ref": null,
            "rendered": "Opening crawl text here",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A New Hope",
              "className": "scrollText",
            },
            "ref": null,
            "rendered": "A New Hope",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1977-05-25",
              "className": "scrollText",
            },
            "ref": null,
            "rendered": "1977-05-25",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
